barretenberg/cpp/src/barretenberg/dsl/acir_format/honk_recursion_constraint.cpp

```diff
@@ -128,14 +126,85 @@ std::array<uint32_t, HonkRecursionConstraint::AGGREGATION_OBJECT_SIZE> create_ho
                 2 * num_frs_comm) %
                    (num_frs_comm + num_frs_fr * UltraFlavor::BATCHED_RELATION_PARTIAL_LENGTH) ==
                0);
-        vkey->log_circuit_size = (input.proof.size() - HonkRecursionConstraint::inner_public_input_offset -
-                                  UltraFlavor::NUM_WITNESS_ENTITIES * num_frs_comm -
-                                  UltraFlavor::NUM_ALL_ENTITIES * num_frs_fr - 2 * num_frs_comm) /
-                                 (num_frs_comm + num_frs_fr * UltraFlavor::BATCHED_RELATION_PARTIAL_LENGTH);
-        vkey->circuit_size = (1 << vkey->log_circuit_size);
-        vkey->num_public_inputs = input.public_inputs.size();
-        vkey->pub_inputs_offset = UltraFlavor::has_zero_row ? 1 : 0;
+        auto log_circuit_size = (input.proof.size() - HonkRecursionConstraint::inner_public_input_offset -
+                                 UltraFlavor::NUM_WITNESS_ENTITIES * num_frs_comm -
+                                 UltraFlavor::NUM_ALL_ENTITIES * num_frs_fr - 2 * num_frs_comm) /
+                                (num_frs_comm + num_frs_fr * UltraFlavor::BATCHED_RELATION_PARTIAL_LENGTH);
+        builder.assert_equal(builder.add_variable(1 << log_circuit_size), key_fields[0].witness_index);
+        builder.assert_equal(builder.add_variable(input.public_inputs.size()), key_fields[1].witness_index);
+        builder.assert_equal(builder.add_variable(UltraFlavor::has_zero_row ? 1 : 0), key_fields[2].witness_index);
+        uint32_t offset = 3;
```

Comment:
could set this to HonkRecursionConstraint::inner_public_input_offset.

Reply:
Actually HonkRecursionConstraint::inner_public_input_offset should probably be moved out of HonkRecursionConstraint.

Reply:
probably don't have to do that in this PR

Reply:
This file needs lot of work/comments